src/vs/workbench/contrib/notebook/browser/diff/cellComponents.ts

```diff
@@ -955,26 +990,123 @@ export class ModifiedCell extends AbstractCellRenderer {
 		super(notebookEditor, cell, templateData, 'full', instantiationService, modeService, modelService, contextMenuService, keybindingService, notificationService, menuService, contextKeyService);
 	}
 
-	initData(): void {
+	styleContainer(container: HTMLElement): void {
 	}
 
-	styleContainer(container: HTMLElement): void {
+	buildBody() {
+		const body = this.templateData.body;
+		this._diffEditorContainer = this.templateData.diffEditorContainer;
+		switch (this.style) {
```

Comment:
This switch statement duplicated above.  We could move this into a new function on the shared parent abstraction can call it from both places?